misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NodeFormattingMetrics.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://wso2.com) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ballerinalang.formatter.core;
+
+import io.ballerinalang.compiler.syntax.tree.BuiltinSimpleNameReferenceNode;
+import io.ballerinalang.compiler.syntax.tree.Node;
+import io.ballerinalang.compiler.syntax.tree.NonTerminalNode;
+import io.ballerinalang.compiler.syntax.tree.ObjectFieldNode;
+import io.ballerinalang.compiler.syntax.tree.SyntaxKind;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+/**
+ * Class that stores the formatting metrics for a particular node.
+ */
+class NodeFormattingMetrics {
+
+    private Node node;
+    private NonTerminalNode parent;
+    private NonTerminalNode grandParent;
+
+    NodeFormattingMetrics(Node node) {
+
+        this.node = node;
+        this.parent = node.parent();
+        if (parent != null) {
+            this.grandParent = parent.parent();
+        } else {
+            this.grandParent = null;
+        }
+        setNodeMetrics();
+    }
+
+    private void setNodeMetrics() {
+        if (node instanceof BuiltinSimpleNameReferenceNode) {
+            handleBuiltinSimpleNameReferenceNode();
+        }
+    }
+
+    boolean handleBuiltinSimpleNameReferenceNode() {
+        boolean addSpaces = true;
+        ArrayList<SyntaxKind> parentNodes = new ArrayList<>(
```

Comment:
Do we need the outer wrapper `new ArrayList<>()`? I think `Arrays.asList()` returns an array-list

Reply:
Fixed.